pkg/controllers/netpol/policy.go

```diff
@@ -85,92 +86,102 @@ func (npc *NetworkPolicyController) syncNetworkPolicyChains(networkPoliciesInfo
 		klog.V(1).Infof("Returned ipset mutex lock")
 	}()
 
-	ipset, err := utils.NewIPSet(false)
-	if err != nil {
-		return nil, nil, err
-	}
-	err = ipset.Save()
-	if err != nil {
-		return nil, nil, err
-	}
-	npc.ipSetHandler = ipset
-
 	activePolicyChains := make(map[string]bool)
 	activePolicyIPSets := make(map[string]bool)
 
+	// for ipFamily, ipset := range npc.ipSetHandlers {
 	// run through all network policies
 	for _, policy := range networkPoliciesInfo {
 
-		// ensure there is a unique chain per network policy in filter table
-		policyChainName := networkPolicyChainName(policy.namespace, policy.name, version)
-		npc.filterTableRules.WriteString(":" + policyChainName + "\n")
+		currentPodIPs := make(map[api.IPFamily][]string)
+		for _, pod := range policy.targetPods {
+			for _, ip := range pod.ips {
+				if utilsnet.IsIPv4String(ip.IP) {
+					currentPodIPs[api.IPv4Protocol] = append(currentPodIPs[api.IPv4Protocol], ip.IP)
+				}
+				if utilsnet.IsIPv6String(ip.IP) {
+					currentPodIPs[api.IPv6Protocol] = append(currentPodIPs[api.IPv6Protocol], ip.IP)
+				}
+			}
+		}
+
+		for ipFamily, ipset := range npc.ipSetHandlers {
+			// ensure there is a unique chain per network policy in filter table
+			policyChainName := networkPolicyChainName(policy.namespace, policy.name, version, ipFamily)
 
-		activePolicyChains[policyChainName] = true
+			npc.filterTableRules[ipFamily].WriteString(":" + policyChainName + "\n")
 
-		currentPodIPs := make([]string, 0, len(policy.targetPods))
-		for ip := range policy.targetPods {
-			currentPodIPs = append(currentPodIPs, ip)
-		}
+			activePolicyChains[policyChainName] = true
 
-		if policy.policyType == kubeBothPolicyType || policy.policyType == kubeIngressPolicyType {
-			// create a ipset for all destination pod ip's matched by the policy spec PodSelector
-			targetDestPodIPSetName := policyDestinationPodIPSetName(policy.namespace, policy.name)
-			npc.createGenericHashIPSet(targetDestPodIPSetName, utils.TypeHashIP, currentPodIPs)
-			err = npc.processIngressRules(policy, targetDestPodIPSetName, activePolicyIPSets, version)
-			if err != nil {
-				return nil, nil, err
+			if policy.policyType == kubeBothPolicyType || policy.policyType == kubeIngressPolicyType {
+				// create a ipset for all destination pod ip's matched by the policy spec PodSelector
+				targetDestPodIPSetName := policyDestinationPodIPSetName(policy.namespace, policy.name, ipFamily)
+				setEntries := make([][]string, 0)
+				for _, podIP := range currentPodIPs[ipFamily] {
+					setEntries = append(setEntries, []string{podIP, utils.OptionTimeout, "0"})
+				}
+				ipset.RefreshSet(targetDestPodIPSetName, setEntries, utils.TypeHashIP)
```

Comment:
Is there a reason not to use `createGenericHashIPSet()` here rather than building the entries yourself? It already accepts `ipFamily` as a parameter, so you would just pass that and it would end up in the correct set.